src/components/AddResourceModal/AddResourceModalV2.jsx

```diff
@@ -92,25 +95,70 @@ export default function AddResourceModalV2(props) {
     });
   };
 
-  const textFieldChangeHandler = e => {
-    setValues(prevValues => {
-      // the address textbox is set with react-hook-form setValue() when user clicks "use my location instead",
-      //  which doesn't fire an event. When that happens, a string is passed instead of an event object, so this
-      // conditional expression handles that
-
-      if (e.target) {
-        return {
-          ...prevValues,
-          [e.target.name]: e.target.value
-        };
+  const debounce = (func, delay) => {
+    let timeoutId;
+    return (...args) => {
+      if (timeoutId) {
+        clearTimeout(timeoutId);
+      }
+      timeoutId = setTimeout(() => {
+        func(...args);
+      }, delay);
+    };
+  };
+
+const debouncedGeocode = debounce(address => {
+  geocodeByAddress(address)
+    .then(results => {
+      if (results.length === 0) {
+        throw new Error('ZERO_RESULTS');
+      }
+      return getLatLng(results[0]);
+    })
+    .then(({ lat, lng }) => {
+      console.log('Successfully got latitude and longitude', { lat, lng });
```

Comment:
Do you think we can remove this console statement? It's useful for local development but doesn't provide value afterward.